pkg/proxy/ipvs/proxier.go

```diff
@@ -856,20 +856,30 @@ func CleanupLeftovers(ipvs utilipvs.Interface, ipt utiliptables.Interface, ipset
 			encounteredError = true
 		}
 	}
-	// Delete dummy interface created by ipvs Proxier.
+	// Delete dummy interface created by ipvs Proxier. Should delete dummy interface once it exist.
 	nl := NewNetLinkHandle()
-	err := nl.DeleteDummyDevice(DefaultDummyDevice)
-	if err != nil {
-		glog.Errorf("Error deleting dummy device %s created by IPVS proxier: %v", DefaultDummyDevice, err)
-		encounteredError = true
+	if found := nl.IsDummyDeviceExist(DefaultDummyDevice); found {
+		err := nl.DeleteDummyDevice(DefaultDummyDevice)
+		if err != nil {
+			glog.Errorf("Error deleting dummy device %s created by IPVS proxier: %v", DefaultDummyDevice, err)
+			encounteredError = true
+		}
 	}
+
 	// Clear iptables created by ipvs Proxier.
 	encounteredError = cleanupIptablesLeftovers(ipt) || encounteredError
 	// Destroy ip sets created by ipvs Proxier.  We should call it after cleaning up
 	// iptables since we can NOT delete ip set which is still referenced by iptables.
 	ipSetsToDestroy := []string{KubeLoopBackIPSet, KubeClusterIPSet, KubeLoadBalancerSet, KubeNodePortSetTCP, KubeNodePortSetUDP,
 		KubeExternalIPSet, KubeLoadBalancerSourceIPSet, KubeLoadBalancerSourceCIDRSet, KubeLoadBalancerMasqSet}
-	for _, set := range ipSetsToDestroy {
+	entryIPSetIn := []string{string(utilipset.HashIPPortIP), string(utilipset.HashIPPort), string(utilipset.HashIPPort), string(utilipset.HashIPPort), string(utilipset.HashIPPort),
```

Comment:
duplicated elements in the slice?

Reply:
Yes, each entry respond to related ipset in `ipSetsToDestroy` slice